### DISCUSSION TITLE
lamp-boot swagger页面接口不显示

### DISCUSSION THREAD

**Discussion by stephen5538** [zuihou/zuihou-admin-cloud#155] [02.提问]
lamp-boot启动以后 swagger没有controller的接口信息
![image](https://user-images.githubusercontent.com/25486527/173310515-4b9ca068-0aa0-4db5-b2c0-affdc4b25ca6.png)

**Comment by zuihou** [comment]
1. lamp-boot 的问题，下次去lamp-boot提issue 或 discussions
2. lamp-boot-server/pom.xml 加入 “权限模块” 的 依赖， 并下载mavenHelper插件，通过搜索确保确实加入成功，能搜索到，才表示加入成功。
<img width="1377" alt="image" src="https://user-images.githubusercontent.com/8500410/173312102-f0a550ee-f14a-4e3d-9ad4-008a0b074f68.png">

3. application.yml 中 扫描“权限模块” 的 controller类所在包
<img width="1219" alt="image" src="https://user-images.githubusercontent.com/8500410/173311868-ae5fa168-d9f6-4d52-9faf-33183dd7fcef.png">

**Reply by stephen5538** [reply]
> 1. lamp-boot 的问题，下次去lamp-boot提issue 或 discussions
> 2. lamp-boot-server/pom.xml 加入 “权限模块” 的 依赖， 并下载mavenHelper插件，通过搜索确保确实加入成功，能搜索到，才表示加入成功。
> 
> <img alt="image" width="1377" src="https://user-images.githubusercontent.com/8500410/173312102-f0a550ee-f14a-4e3d-9ad4-008a0b074f68.png">
> 
> 3. application.yml 中 扫描“权限模块” 的 controller类所在包
> 
> <img alt="image" width="1219" src="https://user-images.githubusercontent.com/8500410/173311868-ae5fa168-d9f6-4d52-9faf-33183dd7fcef.png">

作者你好！
貌似你误会了我的问题，你的回答解决不了我的问题。
因为我本身没有使用你的鉴权方案，所以没有引入UriSecurityPreAuthAspect这个切面类。UriSecurityPreAuthAspect这个类引入与否是导致我问题的直接原因。
我现在解决方式在项目中增加一个Aspect(与你的UriSecurityPreAuthAspect不同里面没有任何逻辑)类并把它注入到spring容器中 接口就可以正常显示，
但是只要把上述的bean删除就显示不了swagger接口信息。
这个问题排查了两天，没有太理解导致这个问题的原因。不知道这是项目的bug还是swagger的bug。
下面我贴一下解决问题代码
![image](https://user-images.githubusercontent.com/25486527/174022036-b61015cb-1d15-409d-a3e5-0ab686731485.png)
![image](https://user-images.githubusercontent.com/25486527/174022076-f738f0b3-01cf-426f-a23c-41323561b919.png)

**Comment by zuihou** [comment]
想不到 UriSecurityPreAuthAspect 和 swagger 有什么关联。 我将 UriSecurityPreAuthAspect 注释也能正常访问swagger

**Reply by stephen5538** [reply]
> 想不到 UriSecurityPreAuthAspect 和 swagger 有什么关联。 我将 UriSecurityPreAuthAspect 注释也能正常访问swagger

具体原因找到了
1.实现了BaseController的类Springboot没有扫描到，把这个实现类就能扫描到。
![image](https://user-images.githubusercontent.com/25486527/174226102-ba95b19b-efb2-4b46-a388-a5e54d31acd4.png)
2.UriSecurityPreAuthAspect 会二次扫描把上面没有扫到的类加到beanfactory里面去
![image](https://user-images.githubusercontent.com/25486527/174226247-774b9bef-51e2-48d4-b58b-c8de9f689e5d.png)
3.把你的boot项目原封不动clone下来倒也没出现这个问题，扫进去
![image](https://user-images.githubusercontent.com/25486527/174226407-cd1b1423-16f6-47de-9a7d-b24dca6469a5.png)
4.自己建一个项目pom中较为精简时出现了上面的问题
![image](https://user-images.githubusercontent.com/25486527/174226598-88c07122-3604-4a4f-98d2-9c15b2607060.png)
不知道作者有没有加了什么限制或者是引入其它依赖，不然照理说没这个道理的，实在太奇怪了
